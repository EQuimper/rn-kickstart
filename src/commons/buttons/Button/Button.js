// @flow

import * as React from 'react';
import { Text, StyleSheet } from 'react-native';
import Touchable from '@appandflow/touchable';

type P = {
  style?: any,
  onPress: () => void,
  children: React.Node,
};

function Button({ style, onPress, children }: P) {
  return (
    <Touchable
      style={[styles.root, style]}
      onPress={onPress}
      feedback="opacity"
    >
      <Text>{children}</Text>
    </Touchable>
  );
}

const styles = StyleSheet.create({
  root: {
    height: 40,
    width: '100%',
    borderRadius: 6,
    backgroundColor: 'red',
    alignItems: 'center',
    justifyContent: 'center'
  },
});

export default Button;
