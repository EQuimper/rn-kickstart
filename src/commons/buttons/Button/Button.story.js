// @flow

import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

import { CenterView } from '../../';
import Button from './Button';

storiesOf('Button', module)
  .addDecorator(getStory => (
    <CenterView style={{ paddingHorizontal: 20 }}>{getStory()}</CenterView>
  ))
  .add('Default', () => (
    <Button variant="primary" onPress={action('clicked-text')}>
      Hello World
    </Button>
  ))
  .add('Disabled', () => (
    <Button disabled variant="primary" onPress={action('clicked-text')}>
      Hello World
    </Button>
  ))
  .add('Loading', () => (
    <Button
      loadingColor="#fff"
      loading
      variant="primary"
      onPress={action('clicked-text')}
    >
      Hello World
    </Button>
  ));
