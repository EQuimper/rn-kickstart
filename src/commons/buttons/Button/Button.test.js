// @noflow

import Button from './Button';

describe('<Button />', () => {
  it('should render correctly', () => {
    expect('hello').toBe('Hello')
  });
});
