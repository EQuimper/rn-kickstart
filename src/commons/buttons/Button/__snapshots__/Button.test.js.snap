// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should be able to change textStyle 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  disabled={false}
  feedback="opacity"
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 8,
        "elevation": 2,
        "height": 40,
        "justifyContent": "center",
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 13,
          "width": 0,
        },
        "shadowOpacity": 0.12,
        "shadowRadius": 25,
        "width": "100%",
      },
      Object {
        "backgroundColor": Object {
          "color": Array [
            233,
            82,
            128,
          ],
          "model": "rgb",
          "valpha": 1,
        },
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "Montserrat-Regular",
          "fontSize": 16,
        },
        Array [
          Object {
            "color": "#fff",
          },
          Object {
            "color": "red",
            "fontSize": 20,
          },
        ],
      ]
    }
  >
    Hello World
  </Text>
</TouchableOpacity>
`;

exports[`<Button /> should be in disabled mode if disabled true 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  disabled={true}
  feedback="opacity"
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 8,
        "elevation": 2,
        "height": 40,
        "justifyContent": "center",
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 13,
          "width": 0,
        },
        "shadowOpacity": 0.12,
        "shadowRadius": 25,
        "width": "100%",
      },
      Object {
        "backgroundColor": Object {
          "color": Array [
            341.7218543046358,
            77.4358974358974,
            92.64705882352942,
          ],
          "model": "hsl",
          "valpha": 1,
        },
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "Montserrat-Regular",
          "fontSize": 16,
        },
        Array [
          Object {
            "color": "#fff",
          },
          undefined,
        ],
      ]
    }
  >
    Hello World
  </Text>
</TouchableOpacity>
`;

exports[`<Button /> should be in disabled mode if loading true and show ActivityIndicator 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  disabled={true}
  feedback="opacity"
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 8,
        "elevation": 2,
        "height": 40,
        "justifyContent": "center",
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 13,
          "width": 0,
        },
        "shadowOpacity": 0.12,
        "shadowRadius": 25,
        "width": "100%",
      },
      Object {
        "backgroundColor": Object {
          "color": Array [
            341.7218543046358,
            77.4358974358974,
            92.64705882352942,
          ],
          "model": "hsl",
          "valpha": 1,
        },
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</TouchableOpacity>
`;

exports[`<Button /> should renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  disabled={false}
  feedback="opacity"
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 8,
        "elevation": 2,
        "height": 40,
        "justifyContent": "center",
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 13,
          "width": 0,
        },
        "shadowOpacity": 0.12,
        "shadowRadius": 25,
        "width": "100%",
      },
      Object {
        "backgroundColor": Object {
          "color": Array [
            233,
            82,
            128,
          ],
          "model": "rgb",
          "valpha": 1,
        },
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "Montserrat-Regular",
          "fontSize": 16,
        },
        Array [
          Object {
            "color": "#fff",
          },
          undefined,
        ],
      ]
    }
  >
    Hello World
  </Text>
</TouchableOpacity>
`;

exports[`<Button /> should throw error if variant or color not passed as props 1`] = `"Button: You must provided at least one of variant or color"`;
